enumerable_methods.rb

```diff
@@ -0,0 +1,265 @@
+      end
+    elsif arg.is_a? Class
+      (0...(to_array.length)).each do |i|
+        return false unless to_array[i].is_a? arg
+      end
+    elsif arg.is_a? Regexp
+      (0...(to_array.length)).each do |i|
+        return false unless to_array[i].match(arg)
+      end
+    elsif !arg.is_a? Regexp and !arg.is_a? Class and !block_given?
+      (0...(to_array.length)).each do |i|
+        return false if to_array[i] != arg
+      end
+    else
+      (0...(to_array.length)).each do |i|
+        return false unless yield(to_array[i])
+      end
+    end
+    true
+  end
+
+  # 5th method
+  def my_any?(arg = nil)
+    if is_a? Range
+      return false if self.end.nil? or self.begin.nil?
+    end
+    to_array = Array(self)
+    return false if to_array.empty?
+
+    if arg.nil? and !block_given?
+      (0...(to_array.length)).each do |i|
+        return true if to_array[i]
+      end
+    elsif arg.is_a? Class
+      (0...(to_array.length)).each do |i|
+        return true if to_array[i].is_a? arg
+      end
+    elsif arg.is_a? Regexp
+      (0...(to_array.length)).each do |i|
+        return true if to_array[i].match(arg)
+      end
+    elsif !arg.is_a? Regexp and !arg.is_a? Class and !block_given?
+      (0...(to_array.length)).each do |i|
+        return true if to_array[i] == arg
+      end
+    else
+      (0...(to_array.length)).each do |i|
+        return true if yield(to_array[i])
+      end
+    end
+    false
+  end
+
+  # 6th method
+  def my_none?(arg = nil)
+    if is_a? Range
+      return false if self.end.nil? or self.begin.nil?
+    end
+    to_array = Array(self)
+    return true if to_array.empty?
+
+    if arg.nil? and !block_given?
+      (0...(to_array.length)).each do |i|
+        return false if to_array[i]
+      end
+    elsif arg.is_a? Class
+      (0...(to_array.length)).each do |i|
+        return false if to_array[i].is_a? arg
+      end
+    elsif arg.is_a? Regexp
+      (0...(to_array.length)).each do |i|
+        return false if to_array[i].match(arg)
+      end
+    elsif !arg.is_a? Regexp and !arg.is_a? Class and !block_given?
+      (0...(to_array.length)).each do |i|
+        return false unless to_array[i] != arg
+      end
+    else
+      (0...(to_array.length)).each do |i|
+        return false if yield(to_array[i])
+      end
+    end
+    true
+  end
+
+  # 7th method
+  def my_count(compare = nil)
+    to_array = Array(self)
+    return puts to_array.length if !block_given? && compare.nil?
+
+    count = 0
+
+    if compare.nil?
+      (0...(to_array.length)).each do |i|
+        count += 1 if yield(to_array[i])
+      end
+    elsif block_given? && !compare.nil?
+      puts 'warning: given block not used'
+      (0...(to_array.length)).each do |i|
+        count += 1 if compare == to_array[i]
+      end
+    else
+      (0...(to_array.length)).each do |i|
+        count += 1 if compare == to_array[i]
+      end
+    end
+    count
+  end
```

Comment:
### My_count

- [ ] When `my_count` is called on a range, it should return the number of items in enum. 

- [ ] When `my_count` is called on an array, it should return the number of items in enum through enumeration. 

* Yours is currently returning **`nil`** for both of these test cases.